File: mobile/src/main/java/com/siliconlabs/bledemo/features/firmware_browser/domain/FirmwareValidation.kt
package com.siliconlabs.bledemo.features.firmware_browser.domain

data class FirmwareValidation(
    val preModel: String,
    val postModel: String,
    val antennaVersion: String,
    val powerVersion: String,
    val afterAntenna: Set<String>? = null,
    val afterPower: Set<String>? = null,
    val afterBoth: Set<String>? = null,
    val scanFilter: ScanFilterConfig? = null,
) {
    fun fieldsToCheck(cardType: CardType?): Set<String> = when (cardType) {
        CardType.ANTENNA -> afterAntenna ?: ALL_FIELDS
        CardType.POWER   -> afterPower   ?: ALL_FIELDS
        CardType.BOTH    -> afterBoth    ?: ALL_FIELDS
        null             -> emptySet()
    }

    companion object {
        val ALL_FIELDS = setOf("post_model", "antenna_version", "power_version")
    }
}

File: OTA_FTP_App/src/domain/firmware_validation.py
"""FirmwareValidation — mirrors the Kotlin data class on the Android side.

Reference: mobile/src/main/java/com/siliconlabs/bledemo/features/firmware_browser/domain/FirmwareValidation.kt
"""
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional, Set

from .scan_filter import ScanFilterConfig


ALL_FIELDS: Set[str] = {"post_model", "antenna_version", "power_version"}


@dataclass
class FirmwareValidation:
    pre_model: str = ""
    post_model: str = ""
    antenna_version: str = ""
    power_version: str = ""
    after_antenna: Optional[Set[str]] = None
    after_power: Optional[Set[str]] = None
    after_both: Optional[Set[str]] = None
    scan_filter: Optional[ScanFilterConfig] = None
